<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>About</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">STAT555-Project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jiuchao/STAT555-Project">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">About</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>STAT555-Project/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.3.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjiuchaoSTAT555Projecttree0c798a99846398eca4d9f49c6079e57464a518abtargetblank0c798a9a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jiuchao/STAT555-Project/tree/0c798a99846398eca4d9f49c6079e57464a518ab" target="_blank">0c798a9</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjiuchaoSTAT555Projecttree0c798a99846398eca4d9f49c6079e57464a518abtargetblank0c798a9a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
working directory clean
</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jiuchao/STAT555-Project/blob/0c798a99846398eca4d9f49c6079e57464a518ab/analysis/about.Rmd" target="_blank">0c798a9</a>
</td>
<td>
jiuchao
</td>
<td>
2019-06-06
</td>
<td>
Add introduction
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jiuchao/STAT555-Project/647b37c457971d64b4175fe2e10638311c34c58c/docs/about.html" target="_blank">647b37c</a>
</td>
<td>
jiuchao
</td>
<td>
2019-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jiuchao/STAT555-Project/89e824252c229801162b79632de3e9271ea1afad/docs/about.html" target="_blank">89e8242</a>
</td>
<td>
jiuchao
</td>
<td>
2019-06-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jiuchao/STAT555-Project/blob/96075ae4f007d3d76cb9c1aa4bca1b32ed3a5317/analysis/about.Rmd" target="_blank">96075ae</a>
</td>
<td>
jiuchao
</td>
<td>
2019-06-06
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="abstract" class="section level1">
<h1><strong>Abstract</strong></h1>
<p>Huntington’s disease (HD) is an inherited neural degenerative disease that will cause gradual neuronal loss in human brain, which further results in movement disorders, psychological changes, and cognitive decline. In this study, induced pluripotent stem cells (iPSCs) from HD patients and the corresponding differentiated neural stem cells (NSCs) were utilized as HD models. Large-scale transcriptome comparison with genetic corrected isogenic control (CT) iPSCs and NSCs showed that HD cells had more difference in differentiated NSC stage than undifferentiated iPSC stage, which is consistent with previous study demonstrating that some disease phenotypes were apparent in HD NSCs but not iPSCs. Categories of differentially expressed genes (DEGs) were also identified in by gene ontology (GO) enrichment analysis. In addition, utilizing such data sets, I was also able to compare the powers of DESeq2 with built-in RLE normalization, edgeR and DESeq2 using TMM normalization. Based on data sets from this paper, I found that DESeq2 using TMM normalization was less powerful than its built-in RLE normalization, the latter of which had similar power with edgeR.</p>
</div>
<div id="introduction" class="section level1">
<h1><strong>Introduction</strong></h1>
<p>Huntington’s Disease (HD) is a genetically inherited disease caused by expansion of CAG (cytosine-adenine-guanine) triplet repeats in the coding region of huntingtin protein (HTT). Such abnormal HTT protein will cause neuronal loss in human brain regions like cortex and striatum (Aronin et al., 1995), which further leads to movement disorders, psychological changes, and cognitive decline (Victorson et al., 2014).</p>
<p>Previous work identified induced pluripotent stem cells (iPSCs) from HD patients and the differentiated neural stem cells (NSCs) as good HD models for mechanistic studies and drug screening (Zhang et al., 2010). iPSCs are pluripotent stem cells that can be reprogrammed from somatic cells, and have the ability to differentiate to any types of somatic cells. NSCs are multipotent stem cells differentiated from pluripotent stem cells, generating mainly neurons and glial cells in the nervous system. Genetic defects of HD-patient-derived iPSCs was successfully corrected previously (An et al., 2012). What is more, characterization of the genetic corrected iPSCs and NSCs showed normalization of cellular phenotypes (An et al., 2012). Therefore, such genetic corrected iPSCs and NSCs serve well as isogenic controls in mechanism study.</p>
<p>Microarray-based gene expression analysis comparing iPSCs from HD and corrected lines previously revealed some useful insights in terms of establishing HD cell models, but also showed that global gene expression remained essentially unchanged at iPSC stage. In contrast, disease phenotypes were manifest in NSC stage concerning cell death, mitochondria dysfunction and susceptibility to growth factor withdrawal (An et al., 2012). However, the characterization with NSCs was only measured by immunohistochemistry and RT-PCR analysis, but not with microarray. To better understand gene expression change caused by HTT protein dysfunction in HD model, large-scale analysis should be performed with not only iPSCs but also NSCs.</p>
<p>In this work, the author utilized HD-patient-derived iPSCs and NSCs as HD models and genetic corrected iPSCs and NSCs as isogenic controls. Transcriptome change measured by RNAseq was performed with both iPSCs and NSCs from HD and control cell lines. This work revealed large-scale gene expression changes caused by HTT protein dysfunction and the gene ontology categories the differentially expressed genes (DEGs) belong to (Ring et al., 2015).</p>
</div>
<div id="background-and-data" class="section level1">
<h1><strong>BACKGROUND AND DATA</strong></h1>
<p>In my project, I used raw read counts obtained by RNAseq from Ring’s work (Ring et al., 2015), which is available in Gene Expression Omnibus (GEO: GSE74201). The data consisted of four conditions: CT iPSC, CT NSC, HD iPSC and HD NSC, with 8 replicates in each condition.</p>
<p>In the original paper, the author did two comparisons: HD versus CT iPSCs and HD versus CT NSCs. Differential expression analysis and weighted co-expression analyses were performed with edgeR and weighted gene coexpression network analysis (WGCNA) packages. After determining clusters of co-regulated genes, the author assessed the gene ontology (GO) annotation of each clusters with DAVID (full names: Database for Annotation, Visualization and Integrated Discovery).</p>
<p>In order to understand gene expression change related to HD, I did the same comparisons: HD versus CT iPSCs and HD versus CT NSCs. The difference was that I used DESeq2 package (Love et al., 2014) to analyze differential expression, and performed GO analysis with Gene Ontology enRIchment anaLysis and visuaLizAtion tool (GOrilla) (Eden et al., 2009).</p>
<p>To assess the power difference among the various differential expression analysis packages, I also compared the power of DESeq2 with built-in RLE normalization, edgeR (Robinson et al., 2010) and DESeq2 with TMM normalization. Other than the two pairwise comparisons I did for studying gene expression changes related to HD, I added one more comparison CT NSC versus CT iPSC for the purpose of power assessment.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
