---
title: "About"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# **Abstract**

Huntington's disease (HD) is an inherited neural degenerative disease that will cause gradual neuronal loss in human brain, which further results in movement disorders, psychological changes, and cognitive decline. In this study, induced pluripotent stem cells (iPSCs) from HD patients and the corresponding differentiated neural stem cells (NSCs) were utilized as HD models. Large-scale transcriptome comparison with genetic corrected isogenic control (CT) iPSCs and NSCs showed that HD cells had more difference in differentiated NSC stage than undifferentiated iPSC stage, which is consistent with previous study demonstrating that some disease phenotypes were apparent in HD NSCs but not iPSCs. Categories of differentially expressed genes (DEGs) were also identified in by gene ontology (GO) enrichment analysis. In addition, utilizing such data sets, I was also able to compare the powers of DESeq2 with built-in RLE normalization, edgeR and DESeq2 using TMM normalization. Based on data sets from this paper, I found that DESeq2 using TMM normalization was less powerful than its built-in RLE normalization, the latter of which had similar power with edgeR. 

# **Introduction**

Huntington's Disease (HD) is a genetically inherited disease caused by expansion of CAG (cytosine-adenine-guanine) triplet repeats in the coding region of huntingtin protein (HTT). Such abnormal HTT protein will cause neuronal loss in human brain regions like cortex and striatum (Aronin et al., 1995), which further leads to movement disorders, psychological changes, and cognitive decline (Victorson et al., 2014).

Previous work identified induced pluripotent stem cells (iPSCs) from HD patients and the differentiated neural stem cells (NSCs) as good HD models for mechanistic studies and drug screening (Zhang et al., 2010). iPSCs are pluripotent stem cells that can be reprogrammed from somatic cells, and have the ability to differentiate to any types of somatic cells. NSCs are multipotent stem cells differentiated from pluripotent stem cells, generating mainly neurons and glial cells in the nervous system. Genetic defects of HD-patient-derived iPSCs was successfully corrected previously (An et al., 2012). What is more, characterization of the genetic corrected iPSCs and NSCs showed normalization of cellular phenotypes (An et al., 2012). Therefore, such genetic corrected iPSCs and NSCs serve well as isogenic controls in mechanism study. 

Microarray-based gene expression analysis comparing iPSCs from HD and corrected lines previously revealed some useful insights in terms of establishing HD cell models, but also showed that global gene expression remained essentially unchanged at iPSC stage. In contrast, disease phenotypes were manifest in NSC stage concerning cell death, mitochondria dysfunction and susceptibility to growth factor withdrawal (An et al., 2012). However, the characterization with NSCs was only measured by immunohistochemistry and RT-PCR analysis, but not with microarray. To better understand gene expression change caused by HTT protein dysfunction in HD model, large-scale analysis should be performed with not only iPSCs but also NSCs.

In this work, the author utilized HD-patient-derived iPSCs and NSCs as HD models and genetic corrected iPSCs and NSCs as isogenic controls. Transcriptome change measured by RNAseq was performed with both iPSCs and NSCs from HD and control cell lines. This work revealed large-scale gene expression changes caused by HTT protein dysfunction and the gene ontology categories the differentially expressed genes (DEGs) belong to (Ring et al., 2015). 

# **BACKGROUND AND DATA**

In my project, I used raw read counts obtained by RNAseq from Ring's work (Ring et al., 2015), which is available in Gene Expression Omnibus (GEO: GSE74201). The data consisted of four conditions: CT iPSC, CT NSC, HD iPSC and HD NSC, with 8 replicates in each condition. 

In the original paper, the author did two comparisons: HD versus CT iPSCs and HD versus CT NSCs. Differential expression analysis and weighted co-expression analyses were performed with edgeR and weighted gene coexpression network analysis (WGCNA) packages. After determining clusters of co-regulated genes, the author assessed the gene ontology (GO) annotation of each clusters with DAVID (full names: Database for Annotation, Visualization and Integrated Discovery).

In order to understand gene expression change related to HD, I did the same comparisons: HD versus CT iPSCs and HD versus CT NSCs. The difference was that I used DESeq2 package (Love et al., 2014) to analyze differential expression, and performed GO analysis with Gene Ontology enRIchment anaLysis and visuaLizAtion tool (GOrilla) (Eden et al., 2009). 

To assess the power difference among the various differential expression analysis packages, I also compared the power of DESeq2 with built-in RLE normalization, edgeR (Robinson et al., 2010) and DESeq2 with TMM normalization. Other than the two pairwise comparisons I did for studying gene expression changes related to HD, I added one more comparison CT NSC versus CT iPSC for the purpose of power assessment.

