---
title: "About"
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

# **ABSTRACT**

Huntington's disease (HD) is an inherited neural degenerative disease that will cause gradual neuronal loss in human brain, which further results in movement disorders, psychological changes, and cognitive decline. In this study, induced pluripotent stem cells (iPSCs) from HD patients and the corresponding differentiated neural stem cells (NSCs) were utilized as HD models. Large-scale transcriptome comparison with genetic corrected isogenic control (CT) iPSCs and NSCs showed that HD cells had more difference in differentiated NSC stage than undifferentiated iPSC stage, which is consistent with previous study demonstrating that some disease phenotypes were apparent in HD NSCs but not iPSCs. Categories of differentially expressed genes (DEGs) were also identified in by gene ontology (GO) enrichment analysis. In addition, utilizing such data sets, I was also able to compare the powers of DESeq2 with built-in RLE normalization, edgeR and DESeq2 using TMM normalization. Based on data sets from this paper, I found that DESeq2 using TMM normalization was less powerful than its built-in RLE normalization, the latter of which had similar power with edgeR. 



