---
title: "project"
author: "jiuchao"
date: "2019-06-06"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---


# Preliminaries

Load the R software libraries required to do the project.

```{r libraries}
library(hexbin)
library(DESeq2)
library(edgeR)
library(qvalue)
library(FactoMineR)
library(gplots)
library(RColorBrewer)
library(pca3d)
```


# 1. RNA-seq Data

## 1.1 Create the RNA-seq count matrix.


```{r readData}
reads = read.csv("analysis/GSE74201_rawReadCounts.csv", header = T, row.names = 1)
dim(reads)
colnames(reads) = paste(rep(c("CT_NSC", "CT_iPSC", "HD_NSC", "HD_iPSC"), each=8), rep(c(1:8), 4))
reads[1:10,]
```
`

***************

## 1.2 Quality assessment

### 1.2.1 Draw a histogram of log2(counts+0.25) for each sample.

```{r, fig.height=15,fig.width=13}
logreads = log2(reads+0.25)
par(mfrow=c(8,4))
for (i in 1:32) hist(logreads[,i],main=colnames(logreads)[i], nclass = 50)
```

### 1.2.2 Use **hexplom** to draw scatterplots of the log2(counts).

```{r, fig.height=13,fig.width=13}
plot(hexplom(logreads[,1:32]))
```


### 1.2.3 Compute the total library size for each of the 12 libraries.

```{r}
lib.size=colSums(reads)
lib.size
```

### 1.2.4 Boxplot of logreads for each sample.

```{r}
boxplot(logreads, outline = TRUE,col = paste(rep(c("Red", "Blue", "orange", "cyan"), each=8)), main = "Boxplot of log2(Gene Expression) of Sample 1 to 32", ylab = "log2 Gene Expression")
```

### 1.2.5 Finally, cluster the samples to get the big picture of whether there are differences among different conditions. 

```{r}
par(mfrow=c(1,1))
dist=dist(t(logreads))
plot(hclust(dist))
```


***************


## 1.2 RNA-seq data filtering and Normalization

### 1.2.1 Discard features with fewer than 10 reads in all the samples as there is no power to detect differential expression analysis for features with very low total counts. Create a new count matrix with genes with at least 10 reads per feature and call it bigReads. 

```{r}
geneSum = rowSums(reads)
bigReads = reads[geneSum>=10,]
nrow(bigReads)
```

### 1.2.2 Normalize data with **TMM** in **edgeR** and compute the dispersion. 

```{r TMMNormalization}
condition=as.factor(paste(rep(c("CT_NSC","CT_iPSC", "HD_NSC", "HD_iPSC"), each = 8)))
condition
d=DGEList(counts=bigReads, group=condition,genes=rownames(bigReads))
d=calcNormFactors(d,method="TMM")
write.csv(d$samples$norm.factors, file ="TMM normalization factors.csv", quote = FALSE)
designmatrix = model.matrix(~0+condition)
colnames(designmatrix) <- levels(condition)
d=estimateDisp(d,designmatrix,robust=TRUE)
plotBCV(d)
```

### 1.2.3 Run **DESeq** and extract the normalized counts for visualization. 

```{r}
samples = colnames(bigReads)
colData = data.frame(samples=samples,condition=condition)
dds = DESeqDataSetFromMatrix(countData=bigReads, colData=colData, design = ~ condition)
dds = DESeq(dds)
res = results(dds)

#Compute normalized counts
nc = counts(dds,normalized=TRUE)
rld = assay(rlog(dds, blind = FALSE))
```

Save the size factors after and plot the dispersion after **DESeq2** normalization.

```{r fig.height=5}
write.csv(dds$sizeFactor, "DESeq2 size factors.csv", quote = FALSE)
plotDispEsts(dds)
```

***************

## 1.3 Check data quality after **DESeq2** normalization

### 1.3.1 Redo the histograms with log2(normalized counts).

```{r, fig.height=15,fig.width=13}
lognc = log2(nc+0.25)
par(mfrow=c(8,4))
for (i in 1:32) hist(lognc[,i],main=colnames(lognc)[i], nclass = 50)
```

### 1.3.2 Redo the scatter plot matrix with log2(normalized counts).

```{r, fig.height=13,fig.width=13}
plot(hexplom(lognc[,1:32]))
```

### 1.3.3 Principle Component Analysis with normalized counts.

```{r}
pca = prcomp(t(nc), scale. = TRUE)
pca_sum = summary(pca)
pca_sum
write.csv(pca_sum$importance,file="summary of pca.csv", quote=FALSE)

#2D demonstration
plot(pca$x[, 1], pca$x[, 2], col = rep(c("red","blue","orange","cyan"),each = 8), main = "Principle Component Analysis", xlab = "PC1:63% variance", ylab = "PC2:8% variace")
plot(pca$x[, 1], pca$x[, 3], col = rep(c("red","blue","orange","cyan"),each = 8), main = "Principle Component Analysis", xlab = "PC1:63% variance", ylab = "PC3: 4% variance")

#3d demonstration
pca3d(pca, group=condition, col = rep(c("red","blue","orange","cyan"),each = 8), axe.titles = c("PC1:63% variance", "PC2:8% variance","PC3:4% variance"), radius = 2)
snapshotPCA3d(file="pca3d.tiff")
```

### 1.3.4 Boxplot demonstration of log2(normalized counts).

```{r}
boxplot(lognc, outline = TRUE,col = paste(rep(c("Red", "Blue", "orange", "cyan"), each=8)), main = "Boxplot of log2(normalized counts) of four conditions", ylab = "log2 Gene Expression")
```

### 1.3.5 Heatmap of sample-to-sample distances.

```{r}
distMatrix = dist(t(lognc))
a = as.dendrogram(hclust(distMatrix))
plot((hclust(distMatrix)))
```
```{r fig.height=9,fig.width=9}
sampleDistMatrix = as.matrix(distMatrix)
colours = colorRampPalette(rev(brewer.pal(9, "Blues")) )(255)
heatmap.2(sampleDistMatrix, trace = "none", col = colours, Rowv = a, Colv = a)
```

 ***************

# 2. Differential Expression Analysis of "HD NSC" and "CT NSC"- **DESeq2**

## 2.1 Extract differential expression results of "HD_NSC vs CT_NSC" from dds.

```{r}
res_CTvsHDNSC = results(dds, contrast = c("condition", "HD_NSC", "CT_NSC"))
```

## 2.2 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_CTvsHDNSC$pvalue, nclass=50, main = "DESeq2", xlab = "p-values")
qDESeq_CTvsHDNSC = qvalue(res_CTvsHDNSC$pvalue)
qDESeq_CTvsHDNSC$pi0
```

## 2.3 repalce genes with ajusted pvalues as "NA" with 1 for DEGs calculation.

```{r}
NA_CTvsHDNSC =sum(is.na(res_CTvsHDNSC$padj))
nc_CTvsHDNSC = nc[,c(1:8, 17:24)]
dim(nc_CTvsHDNSC)
rld_CTvsHDNSC = rld[,c(1:8, 17:24)]
dim(rld_CTvsHDNSC)
res_CTvsHDNSC[is.na(res_CTvsHDNSC$padj),]$padj = rep(1, NA_CTvsHDNSC)
dim(res_CTvsHDNSC)
```

## 2.4 Save rlog transformed data of differentially expressed genes with padj < 0.01 for data visualization later.

```{r}
sum(res_CTvsHDNSC$padj<0.01)
DEGs_CTvsHDNSC = nc[res_CTvsHDNSC$padj<0.01,c(1:8, 17:24)]
rldDEGs_CTvsHDNSC = rld[res_CTvsHDNSC$padj<0.01,c(1:8, 17:24)]
dim(DEGs_CTvsHDNSC)
dim(rldDEGs_CTvsHDNSC)
```

## 2.5 Save upregulated and downregulated genes respectively for GO analysis.

```{r}
upReg_CTvsHDNSC= rownames(nc_CTvsHDNSC[res_CTvsHDNSC$padj<0.01 & res_CTvsHDNSC$log2FoldChange>0,])
length(upReg_CTvsHDNSC)
downReg_CTvsHDNSC = rownames(nc_CTvsHDNSC[res_CTvsHDNSC$padj<0.01 & res_CTvsHDNSC$log2FoldChange<0,])
length(downReg_CTvsHDNSC)
write.table(upReg_CTvsHDNSC,file="upRegulated Genes in NSC.txt", sep="\t",  row.name=FALSE, col.names="Genes",quote=FALSE)
write.table(downReg_CTvsHDNSC,file="downRegulated Genes in NSC.txt", sep="\t",  row.name=FALSE, col.names="Genes",quote=FALSE)
```

## 2.6 Use correlation distance to look at the complete linkage clustering of differentially expressed genes identified by **DESeq2**.

```{r, fig.height=6,fig.width=13}
CorComplete=hclust(as.dist(1-cor(t(rldDEGs_CTvsHDNSC))))
plot(CorComplete,main="Correlation, Complete")
```

## 2.8 Draw heatmap of differentially expressed genes.

```{r, fig.height=9, fig.width=8}
rldDEGs_CTvsHDNSC_M = data.matrix(rldDEGs_CTvsHDNSC)
heatmap.2(rldDEGs_CTvsHDNSC_M,distfun=function(x) as.dist(1-(cor(t(x)))), scale = "row", trace = "none",margins=c(8,8), main = "CT_NSC vs HD_NSC")
```

***************

# 3. Differential Expression Analysis of "HD NSC" and "CT NSC"- **edgR**


## 3.1 Perform **exactTest** of "CT_NSC" and "HD_NSC", draw histogram of p-values, adjust p-values to qvalues and estimate pi0,

```{r}
exact.CTvsHDNSC=exactTest(d,pair=c("CT_NSC","HD_NSC"))
hist(exact.CTvsHDNSC$table$PValue, nclass=50, main = "edgeR", xlab = "p-values")
edgeRq.CTvsHDNSC= qvalue(exact.CTvsHDNSC$table$PValue)
edgeRq.CTvsHDNSC$pi0
```

## 3.2 Get differential expression gene number from **edgeR**, and compare the result with those obtained from **DESeq2**.

```{r}
sum(edgeRq.CTvsHDNSC$qvalues<0.01)
upR_CTvsHDNSC_edgeR = sum(edgeRq.CTvsHDNSC$qvalues<0.01 & exact.CTvsHDNSC$table$logFC>0)
upR_CTvsHDNSC_edgeR
downR_CTvsHDNSC_edgeR = sum(edgeRq.CTvsHDNSC$qvalues<0.01 & exact.CTvsHDNSC$table$logFC<0)
downR_CTvsHDNSC_edgeR
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDNSC$qvalues<0.01,res_CTvsHDNSC$padj<0.01),include="both"),
names=c("edgeR","DESeq2"))
```

***************

# 4. Differential Expression Analysis of "HD NSC" and "CT NSC"- **DESeq2** using normalization factors from **TMM**

## 4.1 Construct *DESeqDataSet*, assign normalization factors obtained from **TMM** to it as size factors. Do **DESeq2** analysis with such pre-existing size factors.

```{r}
dds_2 = DESeqDataSetFromMatrix(countData=bigReads, colData=colData, design = ~ condition)
sizeFactors(dds_2) = 1/d$samples$norm.factors
dds_2 = DESeq(dds_2)
nc_2 = counts(dds_2,normalized=TRUE)
rld_2 = assay(rlog(dds_2, blind = FALSE))
```

## 4.2 Extract differential expression results of "HD_NSC vs CT_NSC" from dds_2.

```{r}
res_CTvsHDNSC2 = results(dds_2, contrast = c("condition", "HD_NSC", "CT_NSC"))
```

## 4.3 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_CTvsHDNSC2$pvalue, nclass=50, main = "DESeq2 with TMM", xlab = "p-values")
qDESeq_CTvsHDNSC2 = qvalue(res_CTvsHDNSC2$pvalue)
qDESeq_CTvsHDNSC2$pi0
```

## 4.4 Repalce genes with ajusted pvalues as "NA" with 1 for DEGs calculation.

```{r}
NA_CTvsHDNSC2 = sum(is.na(res_CTvsHDNSC2$padj))
nc_CTvsHDNSC2 = nc_2[, c(1:8, 17:24)]
dim(nc_CTvsHDNSC2)
rld_CTvsHDNSC2 = rld_2[,c(1:8, 17:24)]
dim(rld_CTvsHDNSC2)
res_CTvsHDNSC2[is.na(res_CTvsHDNSC2$padj),]$padj = rep(1, NA_CTvsHDNSC2)
dim(res_CTvsHDNSC2)
```

## 4.5 Save rlog transformed data of differentially expressed genes with padj < 0.01 for data visualization later.

```{r}
sum(res_CTvsHDNSC2$padj<0.01)
DEGs_CTvsHDNSC2 = nc_CTvsHDNSC2[res_CTvsHDNSC2$padj<0.01,]
rldDEGs_CTvsHDNSC2 = rld_CTvsHDNSC2[res_CTvsHDNSC2$padj<0.01,]
dim(DEGs_CTvsHDNSC2)
dim(rldDEGs_CTvsHDNSC2)
```

## 4.5 Use correlation distance to look at complete linkage clustering of differentially expressed genes identified by **DESeq2** with size factors obtained from **TMM**.

```{r, fig.height=6,fig.width=13}
CorComplete=hclust(as.dist(1-cor(t(rldDEGs_CTvsHDNSC2))))
plot(CorComplete,main="Correlation, Complete")
```

## 4.6 Draw heatmap of differentially expressed genes.

```{r, fig.height=9, fig.width=8}
rldDEGs_CTvsHDNSC_M2 = data.matrix(rldDEGs_CTvsHDNSC2)
heatmap.2(rldDEGs_CTvsHDNSC_M2,distfun=function(x) as.dist(1-(cor(t(x)))), scale = "row", trace = "none",margins=c(8,8))
```

## 4.7 Comparing **edgeR** and **DESeq2** and **DESeq2** using normalization factors from **TMM**.

```{r}
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDNSC$qvalues<0.01,res_CTvsHDNSC2$padj<0.01 ),include="both"),
names=c("edgeR", "DESeq2 TMM"))
vennDiagram(vennCounts(cbind(res_CTvsHDNSC$padj<0.01,res_CTvsHDNSC2$padj<0.01),include="both"),
names=c("DESeq2", "DESeq2 TMM"))
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDNSC$qvalues<0.01,res_CTvsHDNSC$padj<0.01, res_CTvsHDNSC2$padj<0.01 ),include="both"),
names=c("edgeR","DESeq2", "DESeq2 TMM"))
```

***************
 
# 5. Differential Expression Analysis of "HD iPSCs" and "CT iPSCs"- **DESeq2**

## 5.1 Extract differential expression results of "HD_iPSC vs CT_iPSC" from dds.

```{r}
res_CTvsHDiPSC = results(dds, contrast = c("condition", "HD_iPSC", "CT_iPSC"))
```

## 5.2 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_CTvsHDiPSC$pvalue, nclass=50, main = "DESeq2", xlab = "p-values")
qDESeq_CTvsHDiPSC = qvalue(res_CTvsHDiPSC$pvalue)
qDESeq_CTvsHDiPSC$pi0
```

## 5.3 Repalce genes with ajusted pvalues as "NA" with 1 for DEGs calculation.

```{r}
#calculate number of "NA"
NA_CTvsHDiPSC = sum(is.na(res_CTvsHDiPSC$padj))
NA_CTvsHDiPSC
#remove padj == NA
nc_CTvsHDiPSC = nc[,c(9:16, 25:32)]
dim(nc_CTvsHDiPSC)
rld_CTvsHDiPSC = rld[,c(9:16, 25:32)]
res_CTvsHDiPSC[is.na(res_CTvsHDiPSC$padj),]$padj = rep(1, NA_CTvsHDiPSC)
dim(res_CTvsHDiPSC)
```

## 5.4 Save rlog transformed data of differentially expressed genes with padj < 0.01 for data visualization later.

```{r}
sum(res_CTvsHDiPSC$padj<0.01)
DEGs_CTvsHDiPSC = nc_CTvsHDiPSC[res_CTvsHDiPSC$padj<0.01,]
dim(DEGs_CTvsHDiPSC)
rldDEGs_CTvsHDiPSC = rld_CTvsHDiPSC[res_CTvsHDiPSC$padj<0.01,]
dim(rldDEGs_CTvsHDiPSC)
```

## 5.5 Save upregulated and downregulated genes respectively for GO analysis.

```{r}
upReg_TvsHDiPSC= rownames(nc_CTvsHDiPSC[res_CTvsHDiPSC$padj<0.01 & res_CTvsHDiPSC$log2FoldChange>0,])
length(upReg_TvsHDiPSC)
downReg_CTvsHDiPSC = rownames(nc_CTvsHDiPSC[res_CTvsHDiPSC$padj<0.01 & res_CTvsHDiPSC$log2FoldChange<0,])
length(downReg_CTvsHDiPSC)
write.table(upReg_TvsHDiPSC,file="upRegulated Genes CTvsHDiPSC.txt", sep="\t",  row.name=FALSE, col.names="Genes",quote=FALSE)
write.table(downReg_CTvsHDiPSC,file="downRegulated Genes CTvsHDiPSC.txt", sep="\t",  row.name=FALSE, col.names="Genes",quote=FALSE)
```

## 5.6 Use correlation distance to look at complete linkage clustering of differentially expressed genes identified by **DESeq2**.

```{r, fig.height=6,fig.width=13}
CorComplete=hclust(as.dist(1-cor(t(rldDEGs_CTvsHDiPSC))))
plot(CorComplete,main="Correlation, Complete")
```

## 5.7 Draw heatmap of differentially expressed genes.

```{r, fig.height=9, fig.width=8}
rldDEGs_CTvsHDiPSC_M = data.matrix(rldDEGs_CTvsHDiPSC)
heatmap.2(rldDEGs_CTvsHDiPSC_M,distfun=function(x) as.dist(1-(cor(t(x)))), scale = "row", trace = "none",margins=c(8,8), main = "CT_iPSC vs HD_iPSC with DESeq2 ")
```

***************

# 6. Differential Expression Analysis of "HD iPSC" and "CT iPSC"- **edgR**

## 6.1 Perform **exactTest** of "HD_iPSC" and "CT_iPSC", draw histogram of p-values, adjust p-values to qvalues and estimate pi0,

```{r}
exact.CTvsHDiPSC=exactTest(d,pair=c("CT_iPSC","HD_iPSC"))
hist(exact.CTvsHDiPSC$table$PValue, nclass=50, main = "edgeR", xlab = "p-values")
edgeRq.CTvsHDiPSC= qvalue(exact.CTvsHDiPSC$table$PValue)
edgeRq.CTvsHDiPSC$pi0
```


## 6.2 Get differential expression gene number from **edgeR**, and compare the result with those obtained from **DESeq2**.

```{r}
sum(edgeRq.CTvsHDiPSC$qvalues<0.01)
upR_CTvsHDiPSC_edgeR = sum(edgeRq.CTvsHDiPSC$qvalues<0.01 & exact.CTvsHDiPSC$table$logFC>0)
upR_CTvsHDiPSC_edgeR
downR_CTvsHDiPSC_edgeR = sum(edgeRq.CTvsHDiPSC$qvalues<0.01 & exact.CTvsHDiPSC$table$logFC<0)
downR_CTvsHDiPSC_edgeR
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDiPSC$qvalues<0.01,res_CTvsHDiPSC$padj<0.01),include="both"),
names=c("edgeR","DESeq2"))
```

***************

# 7. Differential Expression Analysis of "HD iPSC" and "CT iPSC"- **DESeq2** using normalization factors from **TMM**

## 7.1 Extract differential expression results of "CT_iPSC vs HD_iPSC" from dds_2.

```{r}
res_CTvsHDiPSC2 = results(dds_2, contrast = c("condition", "HD_iPSC", "CT_iPSC"))
```

## 7.2 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_CTvsHDiPSC2$pvalue, nclass=50, main = "DESeq2 with TMM", xlab = "p-values")
qDESeq_CTvsHDiPSC2 = qvalue(res_CTvsHDiPSC2$pvalue)
qDESeq_CTvsHDiPSC2$pi0
```

## 7.3 Remove genes with ajusted pvalues as "NA" from results.

```{r}
# Calculate the number of "NA"
NA_CTvsHDiPSC2 = sum(is.na(res_CTvsHDiPSC2$padj))
NA_CTvsHDiPSC2
#repalce padj == "NA" with 1
nc_CTvsHDiPSC2 = nc_2[,c(9:16, 25:32)]
dim(nc_CTvsHDiPSC2)
rld_CTvsHDiPSC2 = rld_2[,c(9:16, 25:32)]
res_CTvsHDiPSC2[is.na(res_CTvsHDiPSC2$padj),]$padj = rep(1, NA_CTvsHDiPSC2)
dim(res_CTvsHDiPSC2)
```

## 7.4 Save rlog transformed data of differentially expressed genes with padj < 0.01 for data visualization later.

```{r}
sum(res_CTvsHDiPSC2$padj<0.01)
DEGs_CTvsHDiPSC2 = nc_CTvsHDiPSC2[res_CTvsHDiPSC2$padj<0.01,]
dim(DEGs_CTvsHDiPSC2)
rldDEGs_CTvsHDiPSC2 = rld_CTvsHDiPSC2[res_CTvsHDiPSC2$padj<0.01,]
dim(rldDEGs_CTvsHDiPSC2)
```

## 7.5 Use correlation distance to look at complete linkage clustering of differentially expressed genes identified by **DESeq2** with size factors obtained from **TMM**.

```{r, fig.height=6,fig.width=13}
CorComplete=hclust(as.dist(1-cor(t(rldDEGs_CTvsHDiPSC2))))
plot(CorComplete,main="Correlation, Complete")
```

## 7.6 Draw heatmap of differentially expressed genes.

```{r, fig.height=9, fig.width=8}
rldDEGs_CTvsHDiPSC_M2 = data.matrix(rldDEGs_CTvsHDiPSC2)
heatmap.2(rldDEGs_CTvsHDiPSC_M2,distfun=function(x) as.dist(1-(cor(t(x)))), scale = "row", trace = "none",margins=c(8,8))
```

## 7.7 Comparing **edgeR** and **DESeq2** and **DESeq2** using normalization factors from **TMM**.

```{r}
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDiPSC$qvalues<0.01, res_CTvsHDiPSC2$padj<0.01)),include = "both",
names=c("edgeR", "DESeq TMM"))
vennDiagram(vennCounts(cbind(res_CTvsHDiPSC$padj<0.01, res_CTvsHDiPSC2$padj<0.01)),include = "both",
names=c("DESeq2", "DESeq TMM"))
vennDiagram(vennCounts(cbind(edgeRq.CTvsHDiPSC$qvalues<0.01,res_CTvsHDiPSC$padj<0.01, res_CTvsHDiPSC2$padj<0.01)), include = "both",
names=c("edgeR","DESeq2", "DESeq TMM"))
```

***************

# 8. Differential Expression Analysis of "CT NSC" and "CT iPSC"- **DESeq2**

## 8.1 Extract differential expression results of "CT_NSC vs CT_iPSC" from dds.

```{r}
res_iPSCvsNSC = results(dds, contrast = c("condition", "CT_NSC", "CT_iPSC"))
```

## 8.2 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_iPSCvsNSC$pvalue, nclass=50, main = "DESeq2", xlab = "p-values")
qDESeq_iPSCvsNSC  = qvalue(res_iPSCvsNSC$pvalue)
qDESeq_iPSCvsNSC$pi0
```

## 8.3 Replace genes with ajusted pvalues as "NA" with 1 for DEGs calculation.

```{r}
NA_iPSCvsNSC = sum(is.na(res_iPSCvsNSC$padj))
NA_iPSCvsNSC
res_iPSCvsNSC[is.na(res_iPSCvsNSC$padj),]$padj = rep(1, NA_iPSCvsNSC)
dim(res_iPSCvsNSC)
```

## 8.4 calculate the number of differentially expressed genes with padj < 0.01.

```{r}
sum(res_iPSCvsNSC$padj<0.01)
```


***************

# 9. Differential Expression Analysis of "CT NSC" and "CT iPSC"- **edgR**


## 3.1 Perform **exactTest** of "CT NSC" and "CT iPSC", draw histogram of p-values, adjust p-values to qvalues and estimate pi0,

```{r}
exact.iPSCvsNSC=exactTest(d,pair=c("CT_iPSC","CT_NSC"))
hist(exact.iPSCvsNSC$table$PValue, nclass=50, main = "edgeR", xlab = "p-values")
edgeRq.iPSCvsNSC= qvalue(exact.iPSCvsNSC$table$PValue)
edgeRq.iPSCvsNSC$pi0
```

## 9.2 Get differential expression gene number from **edgeR**, and compare the result with those obtained from **DESeq2**.

```{r}
sum(edgeRq.iPSCvsNSC$qvalues<0.01)
vennDiagram(vennCounts(cbind(edgeRq.iPSCvsNSC$qvalues<0.01,res_iPSCvsNSC$padj<0.01),include="both"),
names=c("edgeR","DESeq2"))
```

***************

# 10. Differential Expression Analysis of "CT NSC" and "CT iPSC"- **DESeq2** using normalization factors from **TMM**

## 10.1 Extract differential expression results of "CT_NSC vs CT_iPSC" from dds_2.

```{r}
res_iPSCvsNSC2 = results(dds_2, contrast = c("condition", "CT_NSC", "CT_iPSC"))
```

## 10.2 Plot histogram of p-values and estimate pi0 with qvalue function.

```{r}
hist(res_iPSCvsNSC2$pvalue, nclass=50, main = "DESeq2 with TMM", xlab = "p-values")
qDESeq_iPSCvsNSC2 = qvalue(res_iPSCvsNSC2$pvalue)
qDESeq_iPSCvsNSC2$pi0
```

## 10.3 Remove genes with ajusted pvalues as "NA" from results.

```{r}
NA_iPSCvsNSC2 = sum(is.na(res_iPSCvsNSC2$padj))
NA_iPSCvsNSC2
res_iPSCvsNSC2[is.na(res_iPSCvsNSC2$padj),]$padj = rep(1, NA_iPSCvsNSC2)
dim(res_iPSCvsNSC2)
```

## 10.4 Calculate the number of differentially expressed genes with padj < 0.01.

```{r}
sum(res_iPSCvsNSC2$padj<0.01)
```

## 10.5 Comparing **edgeR** and **DESeq2** and **DESeq2** using normalization factors from **TMM**.

```{r}
vennDiagram(vennCounts(cbind(edgeRq.iPSCvsNSC$qvalues<0.01,res_iPSCvsNSC2$padj<0.01),include="both"),
names=c("edgeR", "DESeq2 TMM"))
vennDiagram(vennCounts(cbind(res_iPSCvsNSC$padj<0.01, res_iPSCvsNSC2$padj<0.01),include="both"),
names=c("DESeq2", "DESeq2 TMM"))
vennDiagram(vennCounts(cbind(edgeRq.iPSCvsNSC$qvalues<0.01,res_iPSCvsNSC$padj<0.01, res_iPSCvsNSC2$padj<0.01 ),include="both"),names=c("edgeR","DESeq2", "DESeq2 TMM"))
```

***************

```{r}
citation("hexbin")
citation("RColorBrewer")
citation("pca3d")
citation("gplots")
citation("FactoMineR")
citation("qvalue")
citation("edgeR")
citation("DESeq2")
```
***************

# SessionInfo

```{r sessionInfo}
sessionInfo()
print(gc())
```

